helper_test.go

```diff
@@ -965,6 +965,43 @@ func createOrganizationWithOptions(t *testing.T, client *Client, options Organiz
 	}
 }
 
+func createOrganizationWithDefaultAgentPool(t *testing.T, client *Client) (*Organization, func()) {
+	ctx := context.Background()
+	org, orgCleanup := createOrganizationWithOptions(t, client, OrganizationCreateOptions{
+		Name:                  String("tst-" + randomString(t)),
+		Email:                 String(fmt.Sprintf("%s@tfe.local", randomString(t))),
+		CostEstimationEnabled: Bool(true),
+	})
+
+	agentPool, agentPoolCleanup := createAgentPool(t, client, org)
+
+	org, err := client.Organizations.Update(ctx, org.Name, OrganizationUpdateOptions{
+		DefaultExecutionMode: String("agent"),
+		DefaultAgentPool:     agentPool,
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return org, func() {
+		// unset execution mode of organization
+		_, err := client.Organizations.Update(ctx, org.Name, OrganizationUpdateOptions{
```

Comment:
I was a little scared to do this, specifically because I know relying on atlas' cleanup logic has caused test flakes in the past, but you're probably right that I was being overprotective in this case. Thanks for the catch!